R/tidiers.R

```diff
@@ -0,0 +1,160 @@
+#' @importFrom generics tidy glance augment
+#' @export
+generics::tidy
+#' @export
+generics::glance
+#' @export
+generics::augment
+
+#' @title Tidying methods for an mmrm model
+#' @description These methods tidy the estimates from an mmrm_tmb object into a
+#' summary
+#' @param x Fitted model object from the mmrm package
+#' @param conf.int If TRUE columns for the lower (conf.low) and upper bounds
+#'   (conf.high) of coefficient estimates are included., Default: FALSE
+#' @param conf.level Defines the range of the confidence internal.
+#'   Only used if conf.int = TRUE, Default: 0.95
+#' @param newdata data.frame, new data frame, Default: NULL
+#' @param se_fit logical, return standard errors of fit,
+#'   Default: interval!='none'
+#' @param interval type of interval calculation. Can be abbreviated,
+#'   Default: c("none", "confidence", "prediction")
+#' @param type.residuals Passed on to residuals.mmrm_tmb().
+#'   Default: c("response", "pearson", "normalized")
+#' @param \dots arguments passed on to predict method
+#' @return All tidying methods return a tibble.
+#'   The structure depends on the method chosen.
+#' @examples
+#'
+#'
+#'   # fit a model
+#'     fit <- mmrm(FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID), data = fev_data)
+#'
+#'   # tidy method
+#'     fit |> tidy()
+#'     fit |> tidy(conf.int = TRUE)
+#'     fit |> tidy(conf.int = TRUE, conf.level = 0.9)
+#'
+#'   # glance method
+#'     fit |> glance()
+#'
+#'   # augment method
+#'     fit |> augment()
+#'     fit |> augment(interval = 'confidence')
+#'     fit |> augment(interval = 'prediction')
+#'
+#'     fit |> augment(type.residuals='pearson')
+#'     fit |> augment(type.residuals='normalized')
+#'
+#' @seealso
+#'  [tibble::as_tibble()], [tibble::enframe()]
+#' @rdname tidiers
+#' @export
+#' @importFrom tibble as_tibble enframe
+tidy.mmrm <- function (x, conf.int = FALSE, conf.level = 0.95) {
+  ret <- summary(x)$coefficients  |> tibble::as_tibble(rownames = "term")
```

Comment:
I still think it would be better for the package code to avoid pipe operator, just for code style consistency. so here e.g.
```suggestion
  coefs_df <- summary(x)$coefficients
  coefs_tibble <- tibble::as_tibble(coefs_df, rownames = "term")
```
etc similarly below

Reply:
removed